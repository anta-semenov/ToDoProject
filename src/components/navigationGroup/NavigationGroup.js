import React from 'react'
import PureRenderMixin from 'react-addons-pure-render-mixin'
require('./NavigationGroup.less')

import NavigationItem from '../navigationItem/NavigationItem'

export default class NavigationGroup extends React.Component {
  constructor(props) {
    super(props)
    this.shouldComponentUpdate = PureRenderMixin.shouldComponentUpdate.bind(this)
  }

  render() {
    return (
      <li className='nav-group'>
        {this.props.title ? <div className='nav-group__title'>{this.props.title}</div> : null}
        <ul className='nav-group__list'>
          {this.props.items.map(item =>
            <NavigationItem
              key={`${item.get('type')}-${item.get('id')}`}
              id={item.get('id')}
              type={item.get('type')}
              title={item.get('title')}
              active={item.get('active')}
              editing={item.get('editing')}
              count={item.get('count')}
              onItemClick={this.props.onItemClick}
              onStopEditing={this.props.onStopEditing}/>
          )}
        </ul>
        {this.props.addNew ?
          <div className='nav-group__add-button' onClick={() => this.props.addNew(this.props.type)}>
            {this.props.addNewTitle}
          </div>:
          null
        }
      </li>
    )
  }
}

NavigationGroup.propTypes = {
  items: React.PropTypes.object.isRequired,
  onItemClick: React.PropTypes.func.isRequired,
  onStopEditing: React.PropTypes.func,

  title: React.PropTypes.string,

  type: React.PropTypes.string,
  addNewTitle : React.PropTypes.string,
  addNew: React.PropTypes.func
}
