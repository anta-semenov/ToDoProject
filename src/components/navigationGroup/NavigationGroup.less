@import '../../styles/variables.less';
@import '../../styles/commonStyles.less';

.nav-group {
  position: relative;
  width: 100%;
  box-sizing: border-box;
  padding: 0;
  margin: 0 0 15px 0;
}

.nav-group__title {
  .small__title();
  margin-top: 15px;
}

.nav-group__list {
  position: relative;
  width: 100%;
  list-style: none;
  margin: 0;
  padding: 0;
}

.nav-group__add-button {
  position: relative;
  padding: 4px 0;
  color: @color-accent;
  font-size: @font-size-small;
  font-weight: 400;
  cursor: pointer;
  transition: @transition-basic;

  &:hover {
    color: darken(@color-accent, 10%);
  }
}
