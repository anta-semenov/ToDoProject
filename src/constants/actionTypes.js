/*
 * Project
 */
export const ADD_PROJECT = 'ADD_PROJECT'
export const REMOVE_PROJECT = 'REMOVE_PROJECT'
export const COMPLETE_PROJECT = 'COMPLETE_PROJECT'
export const EDIT_PROJECT = 'EDIT_PROJECT'
export const REPLACE_PROJECT = 'REPLACE_PROJECT'

/*
 * Task
 */
export const ADD_TASK = 'ADD_TASK'
export const REMOVE_TASK = 'REMOVE_TASK'
export const COMPLETE_TASK = 'COMPLETE_TASK'
export const EDIT_TASK = 'EDIT_TASK'
export const REPLACE_TASK = 'REPLACE_TASK'
export const ADD_TASK_TO_PROJECT = 'ADD_TASK_TO_PROJECT'
export const ADD_TASK_CONTEXT = 'ADD_TASK_CONTEXT'
export const REMOVE_TASK_CONTEXT = 'REMOVE_TASK_CONTEXT'
export const SET_TASK_TODAY = 'SET_TASK_TODAY'
export const START_TASK_TRACKING = 'START_TASK_TRACKING'
export const STOP_TASK_TRACKING = 'STOP_TASK_TRACKING'

/*
 * Context
 */
export const ADD_CONTEXT = 'ADD_CONTEXT'
export const REMOVE_CONTEXT = 'REMOVE_CONTEXT'
export const EDIT_CONTEXT = 'EDIT_CONTEXT'
export const SWITCH_TASK_CONTEXT = 'SWITCH_TASK_CONTEXT'
export const REPLACE_CONTEXT = 'REPLACE_CONTEXT'

/*
* UI State
*/
export const SET_SELECTED_SECTION = 'SET_SELECTED_SECTION'
export const SET_SIDEBAR_SIZE = 'SET_SIDEBAR_SIZE'
export const SET_ACTIVE_ITEM = 'SET_ACTIVE_ITEM'
export const SET_EDITING_SECTION = 'SET_EDITING_SECTION'
export const TOGGLE_TASK_COMPLETED_LATENCY = 'TOGGLE_TASK_COMPLETED_LATENCY'
export const CLEAR_COMPLETED_LATENT_TASKS = 'CLEAR_COMPLETED_LATENT_TASKS'
export const TOGGLE_TASK_LATENCY = 'TOGGLE_TASK_LATENCY'
export const CLEAR_LATENT_TASKS = 'CLEAR_LATENT_TASKS'
export const SET_SYNCING = 'SET_SYNCING'
export const SET_OFFLINE = 'SET_OFFLINE'
export const SET_AUTH_STATUS = 'SET_AUTH_STATUS'
export const SET_AUTH_ERROR_MESSAGE = 'SET_AUTH_ERROR_MESSAGE'
export const SET_PROPERTY = 'SET_PROPERTY'

/*
* Common
*/
export const SET_STATE = 'SET_STATE'
export const LOG_OUT = 'LOG_OUT'
export const REQUEST_AUTH = 'REQUEST_AUTH'
export const RECIEVE_AUTH = 'RECIEVE_AUTH'
export const ERROR_AUTH = 'ERROR_AUTH'

export const REQUEST_DATA = 'REQUEST_DATA'
export const RECIEVE_DATA = 'RECIEVE_DATA'
export const ERROR_DATA = 'ERROR_DATA'

export const UNDO = 'UNDO'
export const REDO = 'REDO'
export const CLEAR_UNDO_REDO = 'CLEAR_UNDO_REDO'
