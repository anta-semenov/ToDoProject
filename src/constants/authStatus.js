export const AUTH_IN_PROGRESS = 'AUTH_IN_PROGRESS'
export const AUTH_SUCESS = 'AUTH_SUCESS'
export const AUTH_ERROR = 'AUTH_ERROR'
export const AUTH_NONE = 'AUTH_NONE'
