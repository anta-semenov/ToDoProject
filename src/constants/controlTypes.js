export const TASK_TITLE_FIELD = 'TASK_TITLE_FIELD'
export const TASK_DESCRIPTION_FIELD = 'TASK_DESCRIPTION_FIELD'
