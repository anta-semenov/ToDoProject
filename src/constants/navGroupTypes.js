export const BASIC = 'BASIC'
export const CONTEXTS = 'CONTEXTS'
export const PROJECTS = 'PROJECTS'
