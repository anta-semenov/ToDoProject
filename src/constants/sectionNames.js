export const INBOX = 'Inbox'
export const NEXT = 'Next'
export const TODAY = 'Today'
export const CONTEXTS = 'Contexts'
export const PROJECTS = 'Projects'
export const SOMEDAY = 'Someday'
