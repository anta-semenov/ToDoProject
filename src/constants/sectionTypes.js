export const PROJECTS = 'PROJECTS'
export const CONTEXTS = 'CONTEXTS'
export const PROJECT = 'PROJECT'
export const CONTEXT = 'CONTEXT'
export const INBOX = 'INBOX'
export const TODAY = 'TODAY'
export const NEXT = 'NEXT'
export const SOMEDAY = 'SOMEDAY'
