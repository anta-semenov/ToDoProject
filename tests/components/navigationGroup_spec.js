import React from 'react'
import { shallow } from 'enzyme'
import { fromJS } from 'immutable'
import * as sectionTypes from '../../src/constants/sectionTypes'
import * as sectionNames from '../../src/constants/sectionNames'
import * as navGroupTypes from '../../src/constants/navGroupTypes'

import NavigationGroup from '../../src/components/navigationGroup/NavigationGroup'

describe('Navigation Group', () => {
  const testGroup = {
    type: navGroupTypes.CONTEXTS,
    title: sectionNames.CONTEXTS,
    items: fromJS([
      {
        type: sectionTypes.CONTEXT,
        title: 'Context 1',
        id: 'cf1sobz3s0o2',
        active: true,
        count: 4,
        editing: false
      },
      {
        type: sectionTypes.CONTEXT,
        title: 'Context 2',
        id: 'cf1sobz3s0o5',
        active: false,
        count: 10,
        editing: false
      }
    ]),
    onItemClick: () => {},
    onStopEditing: () => {},
    addNewTitle: '+ context',
    addNew: () => {}
  }

  it('renders correct component tree', () => {
    const group = shallow(<NavigationGroup {...testGroup} />)
    expect(group).toMatchSnapshot()
  })
})
