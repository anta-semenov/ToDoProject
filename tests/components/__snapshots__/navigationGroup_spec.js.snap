// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Group renders correct component tree 1`] = `
<li
  className="nav-group"
>
  <div
    className="nav-group__title"
  >
    <div
      className="nav-group__title-text"
    >
      Contexts
    </div>
    <div
      className="nav-group__add-button"
      onClick={[Function]}
    />
  </div>
  <ul
    className="nav-group__list"
  >
    <NavigationItem
      active={true}
      changeOrder={[Function]}
      count={4}
      editing={false}
      endDrag={[Function]}
      id="cf1sobz3s0o2"
      index={0}
      key="CONTEXT-cf1sobz3s0o2"
      onItemClick={[Function]}
      onStopEditing={[Function]}
      title="Context 1"
      type="CONTEXT"
    />
    <NavigationItem
      active={false}
      changeOrder={[Function]}
      count={10}
      editing={false}
      endDrag={[Function]}
      id="cf1sobz3s0o5"
      index={1}
      key="CONTEXT-cf1sobz3s0o5"
      onItemClick={[Function]}
      onStopEditing={[Function]}
      title="Context 2"
      type="CONTEXT"
    />
  </ul>
</li>
`;
